sycl/source/device_selector.cpp

```diff
@@ -31,6 +31,24 @@ device device_selector::select_device() const {
 }
 
 int default_selector::operator()(const device &dev) const {
+
+  // Take note of the SYCL_BE environment variable when doing default selection
+  const char *SYCL_BE = std::getenv("SYCL_BE");
+  std::string backend = (SYCL_BE ? SYCL_BE : "");
+  if (backend != "") {
+    const std::string DriverVersion = dev.get_info<info::device::driver_version>();
+    // If we have a cuda device but aren't using PI_CUDA, don't use this device
+    if (DriverVersion.find("CUDA") == std::string::npos && backend == "PI_CUDA") {
+      return -1;
+    }
+    // We can't easily check for an OpenCL device as there is no common string
```

Comment:
would it make sense to check for SPIR support, e.g. `dev.has_extension("cl_khr_spir")` ?

Reply:
We found we could reliably get a string containing OpenCL or CUDA from the platform information, rather than the device information, so won't have to rely on checking the extensions to determine OpenCL support.